x-pack/plugins/gis/public/angular/workspace_controller.js

```diff
@@ -0,0 +1,205 @@
+import { getIsDarkTheme } from '../store/ui';
+import { Inspector } from 'ui/inspector';
+import { inspectorAdapters } from '../kibana_services';
+import { SavedObjectSaveModal } from 'ui/saved_objects/components/saved_object_save_modal';
+import { showSaveModal } from 'ui/saved_objects/show_saved_object_save_modal';
+import { showOptionsPopover } from '../components/top_nav/show_options_popover';
+import { toastNotifications } from 'ui/notify';
+import { getMapReady } from "../selectors/map_selectors";
+
+const REACT_ANCHOR_DOM_ELEMENT_ID = 'react-gis-root';
+
+const app = uiModules.get('app/gis', []);
+
+app.controller('GisWorkspaceController', ($scope, $route, config, breadcrumbState, kbnUrl) => {
+
+  let isLayersListInitializedFromSavedObject = false;
+  const savedWorkspace = $scope.workspace = $route.current.locals.workspace;
+  let isDarkTheme;
+  let unsubscribe;
+
+  getStore().then(store => {
+    handleStoreChanges(store);
+    unsubscribe = store.subscribe(() => {
+      handleStoreChanges(store);
+    });
+
+    // sync store with savedWorkspace mapState
+    // layerList is synced after map has initialized and extent is known.
+    if (savedWorkspace.mapStateJSON) {
+      const mapState = JSON.parse(savedWorkspace.mapStateJSON);
+      store.dispatch(mapExtentChanged({
+        zoom: mapState.zoom,
+        center: mapState.center,
+      }));
+    }
+
+    const root = document.getElementById(REACT_ANCHOR_DOM_ELEMENT_ID);
+    render(
+      <Provider store={store}>
+        <GISApp/>
+      </Provider>,
+      root);
+  });
+
+  function handleStoreChanges(store) {
+    if (isDarkTheme !== getIsDarkTheme(store.getState())) {
+      isDarkTheme = getIsDarkTheme(store.getState());
+      updateTheme();
+    }
+
+    // Part of initial syncing of store from saved object
+    // Delayed until after map is ready so map extent is known
+    if (!isLayersListInitializedFromSavedObject && getMapReady(store.getState())) {
+      isLayersListInitializedFromSavedObject = true;
+      const layerList = savedWorkspace.layerListJSON ? JSON.parse(savedWorkspace.layerListJSON) : [];
+      store.dispatch(replaceLayerList(layerList));
+    }
+  }
+
+  timefilter.on('timeUpdate', dispatchTimeUpdate);
+
+  $scope.$on('$destroy', () => {
+    if (unsubscribe) {
+      unsubscribe();
+    }
+    timefilter.off('timeUpdate', dispatchTimeUpdate);
+    const node = document.getElementById(REACT_ANCHOR_DOM_ELEMENT_ID);
+    if (node) {
+      unmountComponentAtNode(node);
+    }
+  });
+
+  $scope.getWorkspaceTitle = function () {
+    return $scope.workspace.title;
+  };
+  // k7design breadcrumbs
+  // TODO subscribe to store change and change when store updates title
+  breadcrumbState.set([
+    { text: 'Workspace', href: '#' },
+    { text: $scope.getWorkspaceTitle() }
+  ]);
+  config.watch('k7design', (val) => $scope.showPluginBreadcrumbs = !val);
+
+  async function doSave(saveOptions) {
+    const store = await  getStore();
```

Comment:
Store is always defined by this point and exported as a normal value if you wanted to avoid using async